airflow/models/dag.py

```diff
@@ -2711,6 +2712,7 @@ def add_logger_if_needed(ti: TaskInstance):
             secrets_backend_list.insert(0, local_secrets)
 
         execution_date = execution_date or timezone.utcnow()
+        self.validate()
```

Comment:
Why do we need to add this?

Reply:
otherwise we can run `dag.test` on dags that are not valid, that would fail in real life.  i think it's a good thing to have. when adding support here for fail_stop + is_setup, it was a while before i realized it was explicitly forbidden by validate

Reply:
Gotcha, thanks.